packages/mds-agency/request-handlers.ts

```diff
@@ -447,3 +448,47 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
     })
   }
 }
+
+export const registerStop = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+  const stop = req.body
+
+  try {
+    isValidStop(stop)
+    const recorded_stop = await db.writeStop(stop)
+    res.status(201).send(recorded_stop)
+  } catch (err) {
+    if (err instanceof NotFoundError) {
+      res.status(404).send(err.message)
+    }
+    if (err instanceof ValidationError) {
+      res.status(400).send({ error: err })
+    }
+
+    res.status(500).send(new ServerError())
+  }
+}
+
+export const readStop = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+  const { stop_id } = req.params
+  try {
+    const recorded_stop = await db.readStop(stop_id)
+
+    if (!recorded_stop) {
+      throw new NotFoundError()
+    }
+
+    res.status(200).send(recorded_stop)
+  } catch (err) {
+    res.status(404).send(err)
```

Comment:
Are you sure `db.readStop` won't throw an error trying to connect to or read from the db (in which case we'd want to return 500 here)?

Reply:
```
if (!recorded_stop) {
  return res.status(404).send(new NotFoundError())
}
```
Then you can return 500 in the catch.